presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -524,17 +525,24 @@ protected void invalidateTable(String databaseName, String tableName)
     }
 
     @Override
-    public Optional<List<String>> getPartitionNamesByParts(String databaseName, String tableName, List<String> parts)
+    public Optional<List<String>> getPartitionNamesByFilter(
```

Comment:
What is the advantage of having the  `partitionColumns` list in the interface instead of simply pre-filtering the `TupleDomain` to contain only the partition columns predicates?

Reply:
Changed to Map<Column, Domain> as discussed